addons/project/models/project_task.py

```diff
@@ -907,13 +916,7 @@ def _load_records_create(self, vals_list):
             if project_id:
                 self = self.with_context(default_project_id=project_id)
         tasks = super()._load_records_create(vals_list)
-        stage_ids_per_project = defaultdict(list)
-        for task in tasks:
-            if task.stage_id and task.stage_id not in task.project_id.type_ids and task.stage_id.id not in stage_ids_per_project[task.project_id]:
-                stage_ids_per_project[task.project_id].append(task.stage_id.id)
-
-        for project, stage_ids in stage_ids_per_project.items():
-            project.write({'type_ids': [Command.link(stage_id) for stage_id in stage_ids]})
+        tasks._set_stage_on_project_from_task()
```

Comment:
Since `_load_records_create` calls `create`, it is probably no longer needed to call that method here since it will be done in the create method.

Reply:
Indeed, you are correct! Thanks for mentioning.